tests/e2e/restart_test.go

```diff
@@ -137,23 +166,183 @@ func (s *RestartTestData) TestPluginKill(c *C) {
 	c.Assert(err, IsNil, Commentf(out))
 
 	// 5. Start the same container
-	out, err = dockercli.StartContainer(s.config.DockerHosts[1], s.containerName)
+	out, err = dockercli.StartContainer(s.config.DockerHosts[1], s.containerNameList[0])
 	c.Assert(err, IsNil, Commentf(out))
 
 	// 6. Stop the container
-	out, err = dockercli.StopContainer(s.config.DockerHosts[1], s.containerName)
+	out, err = dockercli.StopContainer(s.config.DockerHosts[1], s.containerNameList[0])
 	c.Assert(err, IsNil, Commentf(out))
 
 	// 7. Verify volume detached status
 	status = verification.VerifyDetachedStatus(s.volumeName, s.config.DockerHosts[1], s.config.EsxHost)
 	c.Assert(status, Equals, true, Commentf("Volume %s is still attached", s.volumeName))
 
 	// Cleanup container
-	out, err = dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerName)
+	out, err = dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerNameList[0])
+	c.Assert(err, IsNil, Commentf(out))
+
+	misc.LogTestEnd(c.TestName())
+}
+
+// TestRecoverMountsAfterRestart  - Verify that volume mounts are recovered after
+// docker is restarted and volume is usable by any VM.
+// 1. Start 2 containers using same volume with restart=always
+// 2. Restart docker and wait for containers to start and reference counts to be initialized.
+// 3. Verify volume is attached to the VM
+// 4. Start 1 container using same volume
+// 5. Stop all 3 containers and confirm disk is detached
+// 6. Run container on other host to verify the volume is detached after stopping the containers
+func (s *RestartTestData) TestRecoverMountsAfterRestart(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	c.Skip("Skipping verification of attached status after docker restart")
+	// 1. Run containers with restart-always flag
+	out, err := dockercli.AttachVolumeWithRestart(s.config.DockerHosts[1], s.volumeName, s.containerNameList[0])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Run second container 
+	out, err = dockercli.AttachVolumeWithRestart(s.config.DockerHosts[1], s.volumeName, s.containerNameList[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// 2. Restart docker
+	out, err = dockercli.RestartDocker(s.config.DockerHosts[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	misc.SleepForSec(60)
+
+	// 3. Verify the volume is attached to the VM
+	status, err := dockercli.GetVolumeStatus(s.config.DockerHosts[1], s.volumeName)
+	c.Assert(err, IsNil, Commentf("Failed to fetch status for volume %s", s.volumeName))
+	c.Assert(s.config.DockerHostNames[1], Equals, status["attached_vm"], Commentf("Volume is not attached to VM %s, attached to %s",
+		s.config.DockerHostNames[1], status["attached_vm"]))
+
+	// 4. Run third container
+	out, err = dockercli.AttachVolumeWithRestart(s.config.DockerHosts[1], s.volumeName, s.containerNameList[2])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// 5. Stop the three containers
+	out, err = dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerNameList[0])
```

Comment:
sorry for the confusion it might have cause. 

What I meant instead of invoking 3 calls how about following code block?

```
continerNames := s.containerNameList[0] + " " + s.containerNameList[1] + " "  + s.containerNameList[2] 
out, err = dockercli.RemoveContainer(s.config.DockerHosts[1], continerNames)
c.Assert(err, IsNil, Commentf(out))
```